src/main/java/org/jenkinsci/plugins/github/config/GitHubServerConfig.java

```diff
@@ -184,8 +180,11 @@ public String getName() {
      */
     public String getDisplayName() {
         String n = getName();
-        String a = StringUtils.isBlank(apiUrl) || GITHUB_URL.equals(apiUrl)
-                ? "https://github.com" : StringUtils.removeEnd(apiUrl, "/api/v3");
+        boolean gitHubOrg = StringUtils.isBlank(apiUrl) || GITHUB_URL.equals(apiUrl);
+        if (StringUtils.isBlank(n)) {
+            n = gitHubOrg ? GITHUB_NAME : SCMName.fromUrl(apiUrl, COMMON_PREFIX_HOSTNAMES);
```

Comment:
please avoid one-letter vars next time. In java its very hard to follow code if it has obfuscated variables